aws/resource_aws_dms_endpoint_test.go

```diff
@@ -71,6 +71,11 @@ func TestAccAwsDmsEndpoint_S3(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.bucket_folder", ""),
 					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.bucket_name", "bucket_name"),
 					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.compression_type", "NONE"),
+					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.data_format", "csv"),
+					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.parquet_version", "parquet-1-0"),
+					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.parquet_timestamp_in_millisecond", "false"),
+					resource.TestCheckResourceAttr(resourceName, "s3_settings.0.encryption_mode", "SSE_KMS"),
+					resource.TestCheckResourceAttrPair(resourceName, "s3_settings.0.server_side_encryption_kms_key_id", "data.aws_kms_alias.dms", "target_key_arn"),
```

Comment:
Let's add a separate test to validate the new fields, but checking default values like `parquet_version` can stay here 👍

Reply:
I need some guidance here. I cannot come up with a good name and purpose of the new test. The thing is, I am adding fields to the S3 settings, that is why I extended the test `TestAccAwsDmsEndpoint_S3`. I see there is a separate test for compiling the connection parameter string right below it, but that has a very specific purpose. I do not understand why we need a special test for the new parameters. What makes them so special? The only thing that I found hard is that the change from SSE_KMS encryption to SSE_S3 is allowed, but not the other way around. That is why I chose to start with KMS. Not nice, I must admit.

So, actually, the only thing special here is the encryption setting. Also in regards that it requires the KMS key. Should we have a separate test only for that one, but leave data format and parquet version in the existing test?

Reply:
@anGie44 friendly ping :slightly_smiling_face:

Reply:
Hi @sbrandtb , you bring up a good point..let's get these changes in as-is given the `_S3` has been extended to capture the new fields